### PR TITLE
Get more idiomatic with Groovy, clean up some warnings

### PR DESCRIPTION
**Pull request by abayer**
* JENKINS issue(s):
    * n/a
* Description:
    * Some general code cleanup - I kept meaning to do these things but didn't want to include them in other PRs, so tada.
* Documentation changes:
    * n/a
* Users/aliases to notify:
    * @reviewbybees

### CODE REVIEW

File: pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/Utils.groovy [hunk 1]
```diff
@@ -697,19 +696,19 @@ public class Utils {
...
         for (int x = beginLine; x <= endLine; x++) {
-            String line = sourceUnit.source.getLine(x, null);
+            String line = sourceUnit.source.getLine(x, null)
             if (line == null)
-                throw new AssertionError("Unable to get source line"+x);
+                throw new AssertionError("Unable to get source line"+x)
 
             if (x == endLine) {
-                line = line.substring(0, endLineLastColumn - 1);
+                line = line.substring(0, endLineLastColumn - 1)
```
Comment (rsandell) [inline pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/Utils.groovy]: Could potentially be cleaned a bit more using `str.take` and `str.drop`. Depending on your definition of cleaning :)

File: pipeline-model-definition/src/test/java/org/jenkinsci/plugins/pipeline/modeldefinition/BasicModelDefTest.java [hunk 2]
```diff
@@ -607,9 +614,9 @@ public void skippedStagesForFailure() throws Exception {
                 .logNotContains("I will be skipped", "I also will be skipped", "I have succeeded")
                 .go();
 
-        assertTrue(b.getExecution().getCauseOfFailure() != null);
-
         FlowExecution execution = b.getExecution();
+        assertNotNull(execution);
+        assertTrue(execution.getCauseOfFailure() != null);
```
Comment (rsandell) [inline pipeline-model-definition/src/test/java/org/jenkinsci/plugins/pipeline/modeldefinition/BasicModelDefTest.java]: `assertNotNull` provides better failure messages. Or `assertThat(o, not(null()))`

Comment (abayer) [reply pipeline-model-definition/src/test/java/org/jenkinsci/plugins/pipeline/modeldefinition/BasicModelDefTest.java]: durrr

File: pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/Utils.groovy (hunk 1 above)
Comment (abayer) [reply pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/Utils.groovy]: I have never actually used those yet, so I'll stick with what I'm familiarish with. =)

File: pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/Utils.groovy [hunk 3]
```diff
@@ -442,6 +454,7 @@ public class Utils {
      * @return The list of instances. May be empty.
      */
     @Nonnull
+    @Restricted(NoExternalUse.class)
```
Comment (rsandell) [inline pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/Utils.groovy]: Normally there is no need to restrict package protected methods, but shouldn't hurt. But there is definitely no need to restrict a private method.
otoh this is groovy, so private/public modifiers are just for show :)

Comment (rsandell) [reply pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/Utils.groovy]: Or no modifier in Groovy means `public`? I keep forgetting that :)
Anyway it's a good marker to have so others know to stfo ;)

**Pull request merged**